instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -91,6 +91,8 @@ def collect_request_attributes(scope):
 
 def set_status_code(span, status_code):
     """Adds HTTP response attributes to span using the status_code argument."""
+    if not span.is_recording():
```

Comment:
I wonder if it's worth adding a decorator as a convenience

Reply:
Possibly? I think the repetitive-ness of this task just seems obvious because we just didn't do it in the beginning. However, new instrumentations should be respecting this anyways, so I don't think it's that much overhead.

Reply:
Was thinking a decorator may help folks building future instrumentations group all the actions that should not occur when a span is not recording more cleanly 🤷‍♂️

Reply:
That would be nice. There are some instrumentations that have some interim steps that always must be regardless of is_recording (which occur between some set_attr... methods). A decorator would be difficult to use in those cases.